From: local
Subject: [PATCH] fullscreen adapted for dwm 6.8 + prior patches

Mod+Shift+f toggles monocle layout and bar. Uses pertag layout
index to avoid toggle loop.

--- a/config.def.h
+++ b/config.def.h
@@ -105,6 +105,7 @@
 	{ MODKEY,                       XK_t,      setlayout,      {.v = &layouts[0]} },
 	{ MODKEY,                       XK_f,      setlayout,      {.v = &layouts[1]} },
 	{ MODKEY,                       XK_m,      setlayout,      {.v = &layouts[2]} },
+	{ MODKEY|ShiftMask,             XK_f,      fullscreen,     {0} },
 	{ MODKEY,                       XK_space,  setlayout,      {0} },
 	{ MODKEY|ShiftMask,             XK_space,  togglefloating, {0} },
 	{ MODKEY,                       XK_0,      view,           {.ui = ~0 } },
--- a/dwm.c
+++ b/dwm.c
@@ -228,6 +228,7 @@
 static void setclienttagprop(Client *c);
 static void setfocus(Client *c);
 static void setfullscreen(Client *c, int fullscreen);
+static void fullscreen(const Arg *arg);
 static void setgaps(const Arg *arg);
 static void setlayout(const Arg *arg);
 static void setmfact(const Arg *arg);
@@ -1901,6 +1902,22 @@
 	arrange(selmon);
 }
 
+Layout *last_layout;
+void
+fullscreen(const Arg *arg)
+{
+	if (selmon->pertag->ltidxs[selmon->pertag->curtag][selmon->sellt] != &layouts[2]) {
+		for(last_layout = (Layout *)layouts; last_layout != selmon->lt[selmon->sellt]; last_layout++);
+		setlayout(&((Arg) { .v = &layouts[2] }));
+		if (selmon->showbar)
+			togglebar(arg);
+	} else {
+		setlayout(&((Arg) { .v = last_layout }));
+		if (!selmon->showbar)
+			togglebar(arg);
+	}
+}
+
 void
 setlayout(const Arg *arg)
 {
